eth/handler.go

```diff
@@ -190,9 +190,9 @@ func newHandler(config *handlerConfig) (*handler, error) {
 		}
 	} else {
 		head := h.chain.CurrentBlock()
-		if head.NumberU64() > 0 {
+		if head.NumberU64() > 0 && h.chain.HasState(head.Root()) {
```

Comment:
Can you add more description why this is not needed before but now it's needed (maybe in the comment or commit message)?

Reply:
Added.